packages/core/useScrollParent/index.ts

```diff
@@ -0,0 +1,81 @@
+import type { ComputedRef, MaybeRefOrGetter } from 'vue'
+import type { ConfigurableDocument } from '../_configurable'
+import { computed, toValue } from 'vue'
+import { defaultDocument } from '../_configurable'
+
+export interface UseScrollParentOptions extends ConfigurableDocument {
+  /**
+   * The number of parent levels to lookup
+   *
+   * @default 10
+   */
+  lookupLevel?: number
+  /**
+   * On error callback
+   *
+   * Default log error to `console.error`
+   */
+  onError?: (error: unknown) => void
+}
+
+export type ScrollParent = Element | null | undefined
+
+/**
+ * Reactive nearest scrollable parent of the element
+ *
+ * @see https://vueuse.org/useScrollParent
+ * @param element
+ * @param options
+ */
+export function useScrollParent(
+  element: MaybeRefOrGetter<ScrollParent>,
+  options: UseScrollParentOptions = {},
+): ComputedRef<ScrollParent> {
+  const {
+    document: root = defaultDocument,
+    lookupLevel = 10,
+    onError = (e) => { console.error(e) },
+  } = options
+  const overflowScrollReg = /scroll|auto|overlay/i
+
+  function isElement(node: Element) {
+    const ELEMENT_NODE_TYPE = 1
+    return (
+      node.tagName !== 'HTML'
+      && node.tagName !== 'BODY'
+      && node.nodeType === ELEMENT_NODE_TYPE
+    )
+  }
+
+  function getScrollParent(
+    element: ScrollParent,
+  ) {
+    const rootScrollingElement = root?.scrollingElement || root?.documentElement
+    let node: ScrollParent = element
+    let level = 0
+    while (node && node !== rootScrollingElement && isElement(node as Element) && level < lookupLevel) {
+      const { overflowY } = window.getComputedStyle(node as Element)
+      if (overflowScrollReg.test(overflowY)) {
```

Comment:
What if we're looking in the x axis? Why we can't return an object like { x, y } from the composable to be usable for those cases as well? So the user consumes whichever node it needs.